br/pkg/restore/log_client/client.go

```diff
@@ -1491,24 +1513,36 @@ func (rc *LogClient) GetGCRows() []*stream.PreDelRangeQuery {
 	return rc.deleteRangeQuery
 }
 
-// SaveIDMap saves the id mapping information.
-func (rc *LogClient) SaveIDMap(
+const PITRIdMapBlockSize int = 524288
+
+// saveIDMap saves the id mapping information.
+func (rc *LogClient) saveIDMap(
 	ctx context.Context,
 	sr *stream.SchemasReplace,
 ) error {
-	idMaps := sr.TidySchemaMaps()
-	clusterID := rc.GetClusterID(ctx)
-	metaFileName := metautil.PitrIDMapsFilename(clusterID, rc.restoreTS)
-	metaWriter := metautil.NewMetaWriter(rc.storage, metautil.MetaFileSize, false, metaFileName, nil)
-	metaWriter.Update(func(m *backuppb.BackupMeta) {
-		// save log startTS to backupmeta file
-		m.ClusterId = clusterID
-		m.DbMaps = idMaps
-	})
-
-	if err := metaWriter.FlushBackupMeta(ctx); err != nil {
+	backupmeta := &backuppb.BackupMeta{DbMaps: sr.TidySchemaMaps()}
+	data, err := proto.Marshal(backupmeta)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// clean the dirty id map at first
```

Comment:
is it safe? if process exits abnormally after deleting finished.

Reply:
We have the processes to persist id map:
1. directly get the pitr id map if checkpoint task progress mark that BR has persist pitr id map.
2. if not, generate pitr id map, and 
a. delete the id map at restored ts
b. write the id map at restored ts
c. update the checkpoint task progress to mark that BR has persist pitr id map.

In the external storage, step 2.a and 2.b is replaced with uploading files with override mode. Actually, the atomic is not necessary. That's because there must be failed at the step 2.b or 2.c, which means the pitr id map is incomplete. But just do the step 2.a in the next execution to delete the incomplete pitr id map.

Reply:
Based on id map at `start-ts` and the metakv to generate the new id map at `restored-ts`. So delete the new id map at `restored-ts` is safe because we still can regenerate the same one.